[KIT-76][FE]사용자 내 정보보기 구현

gilmujjang opened the pull request:
-회원가입 입력 폼 수정
-회원가입 성공시, 프로필 화면에서 로드아웃시 메인페이지로 이동
-내 정보 확인 페이지 구현 (리덕스 안쓰고 구현했는데 한번 봐주시면 감사하겠습니다)

gilmujjang committed:
[ADD] 입력폼 사이즈 줄입

gilmujjang committed:
[FIX] 회원가입 질문 목록 수정

gilmujjang committed:
[FIX] 회원가입화면에서 로그인 창 제거

gilmujjang committed:
[ADD] 내정보 확인 페이지 구현

gilmujjang committed:
[ADD] 프로필 페이지 styled component 설정

gilmujjang committed:
[ADD] profile 에서 token 이 없으면 /로 이동, 로그아웃하면 /로 이동

gilmujjang committed:
[ADD] 엔터로 로그인 가능

mayreel0 commented on src/components/LoginDialog/LoginDialog.js:
```diff
@@ -15,7 +15,7 @@ import {
-  const { onLogin, onChange, form, error } = props;
+  const { onLogin, onChange, onEnterPress, form, error } = props;
   const [open, setOpen] = useState(false);
   const dispatch = useDispatch();
```
- components 와 containers 를 분리했던 이유는 두 가지를 분리하기 위해 적용을 했기 때문에
  - container 부분에서 데이터와 상태관리
  - components(presentation 부분) 에서는 어떻게 보여질지(ui, 스타일 등) 
- useState나 dispatch는 props로 받아오는 것이 더 적절하다고 생각합니다.
- https://lunit.gitbook.io/redux-in-korean/basics/usagewithreact#presentational-container

mayreel0 commented on src/components/ProfilePage/ProfilePage.js:
```diff
@@ -3,31 +3,49 @@ import Table from '@material-ui/core/Table';
 import TableCell from '@material-ui/core/TableCell';
 import TableContainer from '@material-ui/core/TableContainer';
-import TableHead from '@material-ui/core/TableHead';
 import TableRow from '@material-ui/core/TableRow';
```
- LoginDialog 에서는 {} 안에서 불러왔지만 여기서는 나누어서 작성한 이유가 있을까요?

gilmujjang replied:
무지성 코딩 죄송합니다 (데헷)

mayreel0 commented on src/components/SignupPage/SignupPage.js:
```diff
@@ -36,14 +41,17 @@ const SignupPage = props => {
           label="ID"
           variant="outlined"
           onChange={handleChange}
         />
```
- 같은 속성을 많이 포함하는 FormTextField가 반복적으로 생긴다면
  - 그 속성들(margin, variant)을 기본으로 포함하고,
  - props 로  나머지 속성을 받는 
  - 작은 컴포넌트를 만들어 코드길이를 줄여보는 생각을 해보는 것도  좋아보이네요.

mayreel0 commented on src/components/SignupPage/SignupPage.js:
```diff
@@ -103,11 +117,18 @@ const SignupPage = props => {
+          <MenuItem value={6}>어머니 성함은?</MenuItem>
+          <MenuItem value={7}>아버지 성함은?</MenuItem>
+          <MenuItem value={8}>가장 좋아하는 색깔은?</MenuItem>
+          <MenuItem value={9}>가장 좋아하는 음식은?</MenuItem>
```
- 이 부분도 Object 나 Array 를 따로 만들어서 작성했어도 괜찮았을것같아요.

mayreel0 commented on src/containers/LoginDialog/LoginDialogContainer.js:
```diff
@@ -47,16 +48,18 @@ const LoginDialogContainer = () => {
+    if (e.key === 'Enter') {
+      onLogin(e);
+    }
   };
```
- 예전 se 매니저페이지 만들 때, 로그인에서 enter 키를 누르지 않아도 form 태그에 onSubmit을 설정하면 따로 설정을 하지 않아도 enter를 누르면 form 데이터가 보내졌었는데 로그인부분은 form 없이 구현하셨나요??

gilmujjang replied:
이 부분은 암만해도 안되네요 😂 se_web_manager 에서 정환님이 작성하신거 참고하는데도 진짜 왜인지 안되는데 한번 봐주시면 감사하겠습니다(진짜 별로 다른것도 없는데 안되네요;;)

mayreel0 replied:
- `<FormField autoComplete="on">` 부분을 `<FormField autoComplete="on" onSubmit={signupSubmit}>` 로 바꾸고 
- `<Button variant="contained" color="primary" onClick={signupSubmit}>` 부분을 `<Button variant="contained" color="primary" type="submit">` 로 바꿔서 한번 시도해보시겠어요??

gilmujjang replied:
그런데도 안되요ㅠㅜ Button 에 onClick 빼면 이젠 버튼도 안먹어요😭

mayreel0 replied:
Button 에 type="submit" 넣어도 안되나요??

mayreel0 commented on src/containers/ProfilePage/ProfilePageContainer.js:
```diff
@@ -1,15 +1,20 @@
+  useEffect(async () => {
+    const myinfo = await api.myinfo({ token });
+    setInfo(myinfo.data.data);
+  }, []);
```
- 사실 이렇게 redux 사용하지 않아도 api 호출하는 게 가능하긴 한데 
  - 먼저 api를 불러올 때 어떤 부분은 saga 를 사용하고, 또 어떤 부분에서는 사용하지 않는 예외를 두는 건 결국 나중에 유지보수가 더 어려워질 수 있다는 생각이 듭니다. 
  - redux 나 redux-saga 말고도 context API 라는 react 자체에서 지원하는 상태관리 방법이 있어서 간단한 경우에는 사용하면 좋지않을까 하고 찾아봤는데 대부분 하나로 일관성 있게 사용 하라고 하더라구요.
- 그래서 일단은 api를 사용하는 부분에선 redux-saga 를 사용하여 진행하는 게 좋아보입니다!

gilmujjang replied:
이부분은 제가 아직 redux-saga 를 확실히 몰라서 편법을 써버렸네요 다시해보겠습니다

mayreel0 commented on src/containers/ProfilePage/ProfilePageContainer.js:
```diff
@@ -1,15 +1,20 @@
+  const token = localStorage.getItem('token');
+  if (token == null) {
+    history.push('/');
+  }
```
- 항상 `==` 대신 `===` 를 사용합시다!

gilmujjang replied:
넵!!

mayreel0 reviewed (changes requested):
- 일단은 API를 사용하는 부분에 대해서는 redux-saga 를 반드시 사용하는 것으로 진행을 하겠습니다.
-  presentational 컴포넌트(/components)와 container 컴포넌트(/containers) 를 확실하게 구분하여 작성하도록 합시다! (저도 이 부분에 대해서는 애매한 부분도 없지않아 있어서... 최대한 나누려고 노력해봅시다!)
- 항상 `==` 대신 `===` 를 사용합시다!

gilmujjang committed:
[REFACTORING] 코드리뷰 내용 수정1
-material ui 컴포넌트 모아서 호출
-회원가입 질문 리스트 객체
-반복되는 css 설정 묶음
-로그인,가입 실패 메세지 표시

gilmujjang committed:
[REFACTORING]  style component 정리 및 유저 타입 객체화

gilmujjang committed:
[REFACTORING] 코드리뷰 결과 수정
-데이터 관리는 Container 에서
-회원가입 실패시 에러문구 표시
-keyPress 없이 enter키로 로그인 가능
-redux 변수 로그인 에러와 가입 에러 분리

gilmujjang committed:
[REFACTORING] 개인정보 조회를 redux-saga 사용
